plugins/skipmap/index.js

```diff
@@ -0,0 +1,159 @@
+        const min = Math.floor(
+          ((options.startTimer - server.layerHistory[0].time) % (1000 * 60 * 60)) / (1000 * 60)
+        );
+        const sec = Math.floor(
+          (options.startTimer - server.layerHistory[0].time) % ((1000 * 60) / 1000)
+        );
+        server.rcon.warn(
+          info.steamID,
+          `Not enough time has passed since the start of the match. Please try again in ${min}min ${sec}s`
+        );
+        return;
+      }
+      // check if enough time remains in the round, if not, inform player
+      if (server.layerHistory[0].time > Date.now() - options.endTimer) {
+        server.rcon.warn(info.steamID, 'Match has progressed too far.');
+        return;
+      }
+
+      // check if enough time has passed since the last vote
+      if (timeLastVote < Date.now() - options.pastVoteTimer) {
+        server.rcon.warn(info.steamID, 'Not enough time has passed since the last vote.');
+        return;
+      }
+
+      // Actual vote
+      playerVotes = {};
+      voteActive = true;
+      votePos = 1;
+      playerVotes[info.steamID] = '+';
+      voteNeg = 0;
+      // Set reminders
+      intervalReminderBroadcasts = setInterval(async () => {
+        await server.broadcast(
+          'A vote to skip the current map has started. Please vote in favour with + or against with -.'
+        );
+        await server.broadcast(`Current counter is: ${votePos} in favour, ${voteNeg} against.`);
+      }, options.reminderInterval);
+
+      // End vote
+      // Disable recording of new votes, stop further broadcasts
+      timeoutVote = setTimeout(() => {
+        voteActive = false;
+        clearInterval(intervalReminderBroadcasts);
+        // Check if enough people voted
+        if (voteNeg + votePos < options.minVoteCount) {
+          server.rcon.broadcast('Not enough people voted for the vote to go through.');
+          return;
+        }
+        if (votePos > voteNeg) {
+          server.rcon.execute('AdminEndMatch');
+        } else {
+          server.rcon.broadcast(
+            `Not enough people voted in favour of skipping the match. ${votePos} voted in favour, ${voteNeg} against.`
+          );
+        }
+        // As a vote happened, stop any further votes from happening until enough time has passed
+        timeLastVote = new Date();
+      }, options.voteDuration);
+    });
+
+    // Clear timeouts and intervals when new game starts
+    server.on(NEW_GAME, (info) => {
+      clearInterval(intervalReminderBroadcasts);
+      clearTimeout(timeoutVote);
+      voteActive = false;
+    });
+
+    // Record votes
+    server.on(CHAT_MESSAGE, async (info) => {
+      if (!voteActive) return;
+
+      // Check if player has voted previously, if yes, remove their vote
+      if (playerVotes[info.steamID]) {
+        if (playerVotes[info.steamID] === '+') votePos--;
+        else voteNeg--;
+      }
+
+      // Record player vote
+      if (playerVotes[info.steamID] === '+') {
```

Comment:
Note for testing: Check whether Squad strips whitespace.

Reply:
Will add `.trim()` to avoid having to test it.